src/SkyDrop.Core/Components/BaseSkyDropHttpClientFactory.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+using System.Net.Http;
+using SkyDrop.Core.DataModels;
+using SkyDrop.Core.Services;
+
+namespace SkyDrop.Core.Components
+{
+    public abstract class BaseSkyDropHttpClientFactory : ISkyDropHttpClientFactory
+    {
+        // todo: nb that this might one day cause issues if we stored too many HttpClient instances e.g. for 100s of portals
+        protected Dictionary<SkynetPortal, HttpClient> HttpClientsPerPortal { get; } = new Dictionary<SkynetPortal, HttpClient>();
+
```

Comment:
The overrides in SkynetPortal allow us to lookup the HttpClient instance with `SkynetPortal(string url)`.

This could equally be a `Dictionary<string, HttpClient>`, but using SkynetPortal makes it a bit more scaleable. You could for example have two HttpClients which point to the same portal but one is private & one authenticates with the user's MySky details.